### PR TITLE
Offline: Add negative tests to illustrate gap in capability around identifying ops submitted in multiple incarnations of the same container

### PR DESCRIPTION
**Pull request by markfields**
## Description

Fixes [AB#7939](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7939)

For the parent Epic, we need to handle the case where a container is serialized, and then multiple incarnations of the same container resubmit the same ops.  Those resubmitted ops will all be on unique clientIds, so the existing op ID logic (which just identifies "local" ops as those matching the current clientId) will not recognize them as duplicates.

This PR adds 3 test cases illustrating this gap.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 1]
```diff
@@ -1915,6 +1916,7 @@ describeCompat("stashed ops", "NoCompat", (getTestObjectProvider, apis) => {
 		}
 	});
 
+	//* Check how this test relates to O3
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:1919]: Oops need to remove this

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 2]
```diff
@@ -1996,6 +1996,113 @@ describeCompat("stashed ops", "NoCompat", (getTestObjectProvider, apis) => {
...
+			obj.clientId = "00000000-0e85-4ea1-983d-3cb72f280701"; // Bogus GUID to simulate reconnect after getPendingLocalState
+			const pendingLocalStateAdjusted = JSON.stringify(obj);
+
+			// load with pending ops with bogus clientId, which makes it impossible (today) to correlate the local ops and they're resent
+			const container2 = await loader.resolve({ url }, pendingLocalStateAdjusted);
+			const dataStore2 = (await container2.getEntryPoint()) as ITestFluidObject;
+			const counter2 = await dataStore2.getSharedObject<SharedCounter>(counterId);
+
+			await provider.ensureSynchronized();
+			assert.strictEqual(counter1.value, 2 * incrementValue);
```
Comment (anthony-murphy) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:2028]: can you document what the correct values should be

Comment (anthony-murphy) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:2028]: i think i would prefer if these test asserted the correct answer, but put a try catch around it, and ensure it breaks today. i think that would make it much clearer what should be happening in these cases, and how fixing it should appear

Comment (markfields) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:2028]: I had thought of that but wanted to also assert the exact expected behavior, asserting what something is !== is not as strong as asserting what it is ===.  Maybe I'll do both

Comment (markfields) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:2028]: I even imagined adding an `it.fails` thing that would make sure the test throws, but not worth it at this moment.

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**